crates/iroha_core/src/query/store.rs

```diff
@@ -37,20 +37,26 @@ use super::cursor::{QueryBatchedErasedIterator, UnknownCursor};
     Decode,
 )]
 pub enum Error {
-    /// Unknown cursor error.
+    /// The query has been dropped from the store.
+    Dropped,
```

Comment:
Might make sense to call it `NotFound` maybe? It's not like we can detect whether the query was in the store once upon a time or not